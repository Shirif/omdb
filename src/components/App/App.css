/* .App {
  text-align: center;
} */

.App {
    min-height: 100vh;
    /* font-size: calc(4px + 2vmin); */
    font-size: 1.5em;
   
    background-image: url(../Pictures/posters_img.png);
    background-color: #000000;
    box-shadow: inset 0 -60px 30px #000000f5;
  }
  