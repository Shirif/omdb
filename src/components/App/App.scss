
.App {
  min-height: 100vh;
  font-size: 1.3em;
  background-image: url(../../Images/posters_img.png);
  background-color: #000000;
  box-shadow: inset 0px 0px 20px 20px #000000;
  background-attachment: fixed;
}
