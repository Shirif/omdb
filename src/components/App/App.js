import React from 'react';
import { useSelector } from "react-redux";
import NavBar from "../NavBar";
import Description from "../Description";
import Spinner from "../Spinner";
import ListMovies from "../ListMovies";
import Pagination from "../Pagination";

import "./App.css";

const App = () => {
  const state = useSelector();
  const movies = state.movies.fetchMovies;
  const loading = state.app.loading;
  const favorite = state.app.favorite;
  const changeMovie = state.movies.changeMovie;
  const totalResults = state.movies.fetchMovies.totalResults;

  return (
    <div className="App">
      <NavBar />
      {changeMovie || movies.Error ? <Description changeMovie={changeMovie} /> : <ListMovies />}
      {loading && <Spinner />}
      {!favorite && <Pagination totalResults={totalResults} />}
    </div>
  );
};

export default App;
