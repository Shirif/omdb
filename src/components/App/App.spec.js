//import * as reactRedux from "react-redux";
import { useSelector } from "react-redux";
import React from "react";
import App from "./App";

const mockAppState = {
  movies: {
    myFavoritMovies: [],
    changeMovie: false,
    fetchMovies: { Error: "Search movie!" },
    searchData: {
      title: "",
      year: "",
      page: "",
    },
  },
  app: {
    loading: false,
    favorite: false,
  },
};

jest.mock("react-redux", () => ({
  ...jest.requireActual("react-redux"),
  useSelector: jest.fn(),
}));

const setUp = () => shallow(<App />);
let component;

describe("should render App components", () => {


  it("should render NavBar, Description, Pagination components", () => {
    useSelector.mockImplementation((callback) => {
      return callback(mockAppState);
    });
    component = setUp();
    expect(component).toMatchSnapshot();
  });

  it("should render ListMovies component", () => {
    useSelector.mockImplementation((callback) => {
      return callback({ ...mockAppState, movies: { ...mockAppState.movies, fetchMovies: {} } });
    });
    component = setUp();
    expect(component).toMatchSnapshot();
  });
  it("should render Spinner component", () => {
    useSelector.mockImplementation((callback) => {
      return callback({ ...mockAppState, app: { ...mockAppState.app, loading: true } });
    });
    component = setUp();
    const wrapper = component.find("Spinner");
    expect(wrapper.length).toBe(1);
  });
});

