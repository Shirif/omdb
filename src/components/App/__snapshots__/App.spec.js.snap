// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render App components should render ListMovies component 1`] = `
<div
  className="App"
>
  <NavBar />
  <ListMovies />
</div>
`;

exports[`should render App components should render NavBar, Description, Pagination components 1`] = `
<div
  className="App"
>
  <NavBar />
  <Description
    changeMovie={false}
  />
  0
</div>
`;
